contracts/Dispatcher.sol

```diff
@@ -548,6 +519,16 @@ contract Dispatcher is IbcDispatcher, Ownable {
         bool hasCommitment = sendPacketCommitment[address(receiver)][packet.src.channelId][packet.sequence];
         require(hasCommitment, 'Packet commitment not found');
 
+        // enforce ack'ed packet sequences always increment by 1 for ordered channels
```

Comment:
The `ack` in IBC term is mixing the concept of a transport layer ack and application layer ack.   So it is dual role. 

- For transport layer, `ack` (such as TCP acks), should be received sequentially.   

- For application layer, `ack` doesn't need to be sequential. 
   -  Some messages doesn't have application ack at all. 
   -  IBC ack can work in `async` mode natively. 
       - These results the ack won't be **always increase by 1**

Reply:
Ordered Acks are enforced for ordered channel. Dapps can choose unordered channel if ordering is not critical to them. What we did here is at the transport layer handled by CoreSC, same as IBC module on IBC chains https://github.com/cosmos/ibc-go/blob/6bfa0b3ea997c1a6c1f99a055935ee170f0ff129/modules/core/04-channel/keeper/packet.go#L443

Reply:
Great clarification.   Yes, I re-checked the spec, it does have similar `functional semantics` requirement for ack to be always generated for ordered channel.